tests/test_java.py

```diff
@@ -718,3 +720,284 @@ def test_java_frames_include_no_semicolons(
+    [
+        (
+            "hotspot-jdk-8",
+            [
+                {
+                    "name": "PrintCodeCache",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "non-default",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SelfDestructTimer",
+                    "type": "intx",
+                    "value": "5",
+                    "origin": "non-default",
+                    "kind": ["product"],
+                },
+            ],
+        ),
+        (
+            "hotspot-jdk-11",
+            [
+                {
+                    "name": "PrintCodeCache",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "environment",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SelfDestructTimer",
+                    "type": "intx",
+                    "value": "5",
+                    "origin": "command line",
+                    "kind": ["product"],
+                },
+            ],
+        ),
+        (
+            "zing",
+            [
+                {
+                    "name": "PrintCodeCache",
+                    "type": "bool",
+                    "value": "true",
+                    "origin": "environment",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SelfDestructTimer",
+                    "type": "intx",
+                    "value": "5",
+                    "origin": "command line",
+                    "kind": ["product"],
+                },
+            ],
+        ),
+    ],
+)
+def test_get_cmdline_and_env_jvm_flags(
+    tmp_path: Path,
+    docker_client: DockerClient,
+    application_docker_image: Image,
+    assert_collapsed: AssertInCollapsed,
+    java_cli_flags: str,
+    java_env_flags: str,
+    expected_flags: List[Dict[str, Union[str, List[str]]]],
+    application_pid: int,
+    application_image_tag: str,
+) -> None:
+    """
+    Tests that we can profile a Java app that runs with non-java "comm", by reading the argv0 instead.
+    """
+    with make_java_profiler(
+        frequency=99,
+        storage_dir=str(tmp_path),
+        java_async_profiler_mode="cpu",
+        java_jvm_flags_to_retrieve="SelfDestructTimer,PrintCodeCache",
+    ) as profiler:
+        # When running a container manually we can't use application_pid fixture as it will come from the fixture
+        # container and not from the manully started one
+        with _application_docker_container(
+            docker_client,
+            application_docker_image,
+            [],
+            [],
+            application_docker_command=[
+                "bash",
+                "-c",
+                f"export JAVA_TOOL_OPTIONS={java_env_flags}" ";" f"java {java_cli_flags} -jar Fibonacci.jar",
```

Comment:
```suggestion
                f"export JAVA_TOOL_OPTIONS={java_env_flags}" ";" f"exec java {java_cli_flags} -jar Fibonacci.jar",
```

if you do that you won't need the `find_application_pid`. You can keep the exporting fo it to a function, though.